api/include/opentelemetry/trace/span_id.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace opentelemetry {
+namespace trace {
+
+class SpanId {
+ public:
+  // The size in bytes of the SpanId.
+  static constexpr size_t kSize = 8;
+
+  // An invalid SpanId (all zeros).
+  SpanId() : rep_{0} {}
+
+  // Creates a SpanId by copying the first kSize bytes from the buffer.
+  explicit SpanId(const uint8_t* buf);
```

Comment:
Do we require raw ptr here? If it's meant to be an array, then we can't pass just a pointer to it:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#i13-do-not-pass-an-array-as-a-single-pointer

Can we use pass by const reference of std::vector<uint8_t> instead? That should work well with C++11 compiler.

Reply:
I think per docs/requirements.md we should avoid `std::vector`.

How else can we hint to the compiler about the length of the input?